src/main/java/com/geespring/marketapi/service/UserService.java

```diff
@@ -0,0 +1,80 @@
+package com.geespring.marketapi.service;
+
+import com.geespring.marketapi.model.User;
+import com.geespring.marketapi.repository.UserRepository;
+import com.geespring.marketapi.util.TimeUtil;
+import com.geespring.marketapi.util.exceptions.DataNotFoundException;
+import com.geespring.marketapi.util.exceptions.DuplicateDataFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class UserService {
+
+    /**
+     * Represents the Account Repository Provider
+     */
+    @Autowired
+    private final UserRepository repository;
+
+    /**
+     * Represents the TimeUtil Component Instance
+     */
+    @Autowired
+    private final TimeUtil timeUtil;
+
+    /**
+     * Creates and saves the requested {@link User} object into the database
+     * @param user
+     * @return the requested account object
+     */
+    public User create(User user) {
+        if (repository.findByEmailAddress(user.getEmail()).isPresent()) {
```

Comment:
This is unreliable because a race condition between two running instances of the same API can allow this to pass when it shouldn't. If you set up your database schema with the correct uniqueness constraints, the database will verify this before final insertion and give you an exception if a user with the email already exists.